lsp/nls/src/linearization/interface.rs

```diff
@@ -24,7 +24,12 @@ impl ResolutionState for Resolved {}
 /// Can be extended later to represent Contracts, Records, etc.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum TermKind {
-    Declaration(Ident, Vec<ItemId>, ValueState),
+    Declaration {
+        id: Ident,
+        usages: Vec<ItemId>,
+        value: ValueState,
+        pattern_bindings: Option<Vec<Ident>>,
```

Comment:
Then, I think you should add what you just explained in an answer to my comment directly in the code, so that I'm not the only one to enjoy it :slightly_smiling_face: 

As a side note, `pattern_bindings` doesn't reflect what this variables really represent IMHO. There is a notion of `path`, so `path` should probably be in the name. There aren't bindings, because none but potentially the last element of the path is actually bound in the environment.

Reply:
Agreed. Thanks! 🙂